doc/modules/cross_validation.rst

```diff
@@ -856,3 +856,53 @@ Cross validation and model selection
 Cross validation iterators can also be used to directly perform model
 selection using Grid Search for the optimal hyperparameters of the
 model. This is the topic of the next section: :ref:`grid_search`.
+
+.. _permutation_test_score:
+
+Permutation test score
+======================
+
+:func:`~sklearn.model_selection.permutation_test_score` offers another way
+to evaluate the performance of classifiers. It provides a permutation-based
+p-value, which represents how likely an observed performance of the
+classifier would be obtained by chance. The null hypothesis in this test is
+that the features and labels are independent.
```

Comment:
The null hypothesis is that the classifier fails leverage any statistical dependency between X and y to make correct predictions on left out data.

Reply:
Hmm the [Ojala and Garriga paper](http://www.jmlr.org/papers/volume11/ojala10a/ojala10a.pdf) says:
> The hypothesis assumes that the data X and the labels y are independent, that is, p(X, y) = p(X)p(y).
(pg 1839)

but you're right that we do use performance of the classifier on left out data for our null distribution. But by permuting the labels we do remove any dependency between X and y - so I think this should be part of of the null hypothesis.

Reply:
I think that generating random X and y with broken dependency is just a way to ensure that "the classifier fails to leverage any statistical dependency between X and y to make correct predictions on left out data."

The statistical test is not a test about the data alone, it's a test about how well the model can (or cannot) extract useful things from the data.

Reply:
Yes, that makes sense, thanks!